tests/unit/avro/test_serde.py

```diff
@@ -77,6 +77,15 @@ def test_serialize_bad_datum_empty():
     with pytest.raises(DatumTypeException):
         serialize(USER_SCHEMA, [user])
 
+def test_serialize_bad_type_for_field():
```

Comment:
Could you add a test to show how / when we can expect Value Error vs Type Error vs Validation Error ?